---
 Driver/ToolChains.cpp         |    5 +++--
 Frontend/InitHeaderSearch.cpp |    5 +++++
 2 files changed, 8 insertions(+), 2 deletions(-)

ss diff --unidirectional-new-file -ur clang-2.9.orig//tools/clang/lib/Driver/ToolChains.cpp clang-2.9/tools/clang/lib/Driver/ToolChains.cpp
--- clang-2.9.orig//tools/clang/lib/Driver/ToolChains.cpp	2011-06-23 15:38:16.000000000 +0200
+++ clang-2.9/tools/clang/lib/Driver/ToolChains.cpp	2011-06-23 15:32:54.000000000 +0200
@@ -28,6 +28,7 @@
 #include "llvm/Support/raw_ostream.h"
 #include "llvm/Support/Path.h"
 #include "llvm/Support/system_error.h"
+#include "clang/Debian/debian_path.h"
 
 #include <cstdlib> // ::getenv
 
@@ -1460,8 +1461,8 @@
                                "4.2"};
   std::string Base = "";
   for (unsigned i = 0; i < sizeof(GccVersions)/sizeof(char*); ++i) {
-    std::string Suffix = GccTriple + "/" + GccVersions[i];
-    std::string t1 = "/usr/lib/gcc/" + Suffix;
+      std::string Suffix = std::string(DEB_HOST_GNU_TYPE_TRIPLET) + "/" + GccVersions[i];
+      std::string t1 = "/usr/lib/gcc/" + Suffix;
     if (!llvm::sys::fs::exists(t1 + "/crtbegin.o", Exists) && Exists) {
       Base = t1;
       break;
Seulement dans clang-2.9.orig//tools/clang/lib/Driver: ToolChains.cpp.orig
--- clang-2.9.orig/tools/clang/lib/Frontend/InitHeaderSearch.cpp	2011-08-11 12:53:16.000000000 +0200
+++ clang-2.9/tools/clang/lib/Frontend/InitHeaderSearch.cpp	2011-08-11 14:04:37.000000000 +0200
@@ -33,6 +33,7 @@
   #define WIN32_LEAN_AND_MEAN 1
   #include <windows.h>
 #endif
+#include "clang/Debian/debian_path.h"
 using namespace clang;
 using namespace clang::frontend;
 
@@ -450,6 +451,10 @@
     break;
   }
 
+  std::string pathInclude = "/usr/include/" + std::string(DEB_HOST_MULTIARCH_TRIPLET) + "/";
+  AddPath(pathInclude, System, true, false, false);
+
+
   // Builtin includes use #include_next directives and should be positioned
   // just prior C include dirs.
   if (HSOpts.UseBuiltinIncludes) {
